src/lib/core/types/study.ts

```diff
@@ -231,3 +232,9 @@ export const StudyMetadata = t.intersection([
     rootEntity: StudyEntity,
   }),
 ]);
+
+export type ExtendedField = Field & {
+  precision: number;
+  variableName: string;
+  isFeatured?: boolean;
+};
```

Comment:
It might also be worth declaring a type alias like `type FieldWithMetadata = Field | ExtendedField`, and using that new alias in place of `Field | ExtendedField`. That should help if we ever change the field metadata.

(I'm not sold on the name `FieldWithMetadata`.)